SpeedtestPlugin/Classes/ByteBitSize.cs

```diff
@@ -0,0 +1,110 @@
+    public class ByteSize : _IgnoreIgnoreIgnore.ByteBitSizeHelper<ByteSize.SizeUnit>
+    {
+        static ByteSize()
+        {
+            UnitIncrement = 1024;
+            UnitDisplayChars = 2;
+        }
+        public static Int64 BytesFromTB(Double value) => SmallestUnitFromUnitValue(value, SizeUnit.TB);
+        public static Int64 BytesFromGB(Double value) => SmallestUnitFromUnitValue(value, SizeUnit.GB);
+        public static Int64 BytesFromMB(Double value) => SmallestUnitFromUnitValue(value, SizeUnit.MB);
+        public static Int64 BytesFromKB(Double value) => SmallestUnitFromUnitValue(value, SizeUnit.KB);
+        public static Int64 BytesFromBits(Double value) => (Int64)Math.Round(value / 8);
+        public enum SizeUnit { B = 1, KB = 2, MB = 3, GB = 4, TB = 5 };
+        //have to do this to make sure our constructor gets claled
+        public static Int64 SmallestUnitFromUnitValue(Double unit_amt, SizeUnit unit) => _SmallestUnitFromUnitValue(unit_amt, unit);
+        public static Int64 SizeFromHumanReadable(String sizeStrWithType) => _SizeFromHumanReadable(sizeStrWithType);
+        public static Int64 SizeFromHumanReadable(String sizeStr, String sizeType) => _SizeFromHumanReadable(sizeStr, sizeType);
+        public static Int64 SizeFromHumanReadable(String sizeStr, SizeUnit sizeType) => _SizeFromHumanReadable(sizeStr, sizeType);
+        public static String HumanReadable(Double len, Int32 decimalPlaces = 2) => _HumanReadable(len, decimalPlaces);
+    }
+    public class BitSize : _IgnoreIgnoreIgnore.ByteBitSizeHelper<BitSize.SizeUnit>
+    {
+        static BitSize()
+        {
+            UnitIncrement = 1000;
+            UnitDisplayChars = 4;
+        }
+        public static Int64 BitsFromTbits(Double value) => SmallestUnitFromUnitValue(value, SizeUnit.tbit);
+        public static Int64 BitsFromGbits(Double value) => SmallestUnitFromUnitValue(value, SizeUnit.gbit);
+        public static Int64 BitsFromMbits(Double value) => SmallestUnitFromUnitValue(value, SizeUnit.mbit);
+        public static Int64 BitsFromKbits(Double value) => SmallestUnitFromUnitValue(value, SizeUnit.bit);
+        public static Int64 BitsFromBytes(Double value) => (Int64)Math.Round(value * 8);
+        public static Int64 SmallestUnitFromUnitValue(Double unit_amt, SizeUnit unit) => _SmallestUnitFromUnitValue(unit_amt, unit);
+        public static Int64 SizeFromHumanReadable(String sizeStrWithType) => _SizeFromHumanReadable(sizeStrWithType);
+        public static Int64 SizeFromHumanReadable(String sizeStr, String sizeType) => _SizeFromHumanReadable(sizeStr, sizeType);
+        public static Int64 SizeFromHumanReadable(String sizeStr, SizeUnit sizeType) => _SizeFromHumanReadable(sizeStr, sizeType);
+        public static String HumanReadable(Double len, Int32 decimalPlaces = 2) => _HumanReadable(len, decimalPlaces);
+        public enum SizeUnit { bit = 1, kbit = 2, mbit = 3, gbit = 4, tbit = 5 }
+    }
+
+}
+namespace _IgnoreIgnoreIgnore

```

Comment:
I'm not a bug friend of such namespaces. Is there any reason for this? If no, could you please change this?